aiida/engine/processes/control.py

```diff
@@ -0,0 +1,206 @@
+
+    :return: A list of process nodes of active processes.
+    """
+    filters = CalculationQueryBuilder().get_filters(process_state=('created', 'waiting', 'running'), paused=True)
+    builder = QueryBuilder().append(ProcessNode, filters=filters)
+    return builder.all(flat=True)
+
+
+def play_processes(
+    processes: list[ProcessNode] | None = None,
+    all_entries: bool = False,
+    timeout: float = 5.0,
+    wait: bool = False
+) -> None:
+    """Play (unpause) paused processes.
+
+    .. note:: Requires the daemon to be running, or processes will be unresponsive.
+
+    :param processes: List of processes to play.
+    :param all_entries: Play all paused processes.
+    :param timeout: Raise a ``ProcessTimeoutException`` if the process does not respond within this amount of seconds.
+    :param wait: Set to ``True`` to wait for process response, for ``False`` the action is fire-and-forget.
+    :raises ``ProcessTimeoutException``: If the processes do not respond within the timeout.
+    """
+    if processes and all_entries:
+        raise ValueError('cannot specify processes when `all_entries = True`.')
+
+    if not processes and all_entries:
+        processes = get_active_processes()
+
+    if not processes:
+        raise ValueError('no processes specified and `all_entries = False`.')
+
+    controller = get_manager().get_process_controller()
+    _perform_actions(processes, controller.play_process, 'play', 'playing', 'played', wait, timeout)
+
+
+def pause_processes(
+    processes: list[ProcessNode] | None = None,
+    all_entries: bool = False,
+    timeout: float = 5.0,
+    wait: bool = False
+) -> None:
+    """Pause running processes.
+
+    .. note:: Requires the daemon to be running, or processes will be unresponsive.
+
+    :param processes: List of processes to play.
+    :param all_entries: Play all paused processes.
+    :param timeout: Raise a ``ProcessTimeoutException`` if the process does not respond within this amount of seconds.
+    :param wait: Set to ``True`` to wait for process response, for ``False`` the action is fire-and-forget.
+    :raises ``ProcessTimeoutException``: If the processes do not respond within the timeout.
+    """
+    if processes and all_entries:
+        raise ValueError('cannot specify processes when `all_entries = True`.')
+
+    if not processes and all_entries:
+        processes = get_active_processes()
+
+    if not processes:
+        raise ValueError('no processes specified and `all_entries = False`.')
+
+    controller = get_manager().get_process_controller()
+    _perform_actions(processes, controller.pause_process, 'pause', 'pausing', 'paused', wait, timeout)
+
+
+def kill_processes(processes: list[ProcessNode], timeout: float = 5.0, wait: bool = False) -> None:
+    """Kill running processes.
+
+    .. note:: Requires the daemon to be running, or processes will be unresponsive.
+
+    :param processes: List of processes to play.
```

Comment:
why no `all_entries` here?

Can certainly imagine someone wanting to kill all running processes

Reply:
Could be dangerous? It isn't there currently in the API, and we haven't really had anyone ask for it. But happy to add it if you think that's fine